mllib/pom.xml

```diff
@@ -142,20 +142,20 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>dev.ludovic.netlib</groupId>
+      <artifactId>blas</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>dev.ludovic.netlib</groupId>
+      <artifactId>lapack</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>dev.ludovic.netlib</groupId>
+      <artifactId>arpack</artifactId>
+    </dependency>
+
   </dependencies>
-  <profiles>
```

Comment:
What do your libraries pull in that this profile might have? it's disabled on purpose as we can't include what it pulls in in the release itself. (We also probably need to leave the profile empty at least to avoid bulid errors on a missing profile)

Reply:
It only pulls `com.github.fommil.netlib`. The long term goal is to get rid of this dependency but it's going to require quite a lot of work still.

What I don't understand is how can Spark work today if `com.github.fommil.netlib` is not installed? Are there other packages that can be dropped in place and which can seamlessly replace it? `com.github.fommil.netlib` is unconditionally added as a dependency in [graphx/pom.xml](https://github.com/apache/spark/blob/master/graphx/pom.xml#L63), and that is why Spark functions even when `-Pnetlib-lgpl` isn't defined. (The package is going to be downloaded, the jar is going to be passed to the compiler, and it's going to be on the classpath when running tests and benchmarks).

Reply:
I believe that netlib has two parts, the core Java interface, and a fuller package including LGPL components. I can work with only the core - falls back to JVM code. The artifacts behind the profile are the complete "all" artifact. I think we'd have to follow the same. Anything importing "all" needs to be behind the profile. You can check the result by running `dev/test-dependencies.sh --replace-manifest` (which you'll need to do anyway) to see if the changes in the transitive dependencies have included the "all" artifact.

Reply:
I have removed the dependency on `com.github.fommil.netlib:all` in `dev.ludovic.netlib`, and replaced it with `com.github.fommil.netlib:core` per your previous message. It's then still up to the Spark user to specify `-Pnetlib-lgpl` to get the native wrapper in `com.github.fommil.netlib` or not.

In the longer term, I'll add JNI-based native wrappers to `dev.ludovic.netlib` so that we wouldn't have to rely on `com.github.fommil.netlib:all` at all. These native wrappers would only depend on a native BLAS implementation (OpenBLAS or Intel MKL for example) to be already installed on the machine, but wouldn't ship any GPL/LGPL code.

Reply:
I also moved this dependency to https://github.com/apache/spark/pull/32253/files/#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R3489-R3499